deployment/vm/manager_test.go

```diff
@@ -161,6 +161,59 @@ var _ = Describe("Manager", func() {
 			Expect(fakeVMRepo.UpdateCurrentCID).To(Equal("fake-vm-cid"))
 		})
 
+		Context("when current IP is specified", func() {
+			BeforeEach(func() {
+				fakeVMRepo.SetFindCurrentIPBehavior("10.10.1.3", true, nil)
+				_, _, err := fakeVMRepo.FindCurrentIP()
+				Expect(err).ToNot(HaveOccurred())
+
+				fakeVMRepo.SetFindCurrentAgentIdBehavior("fake-existed-agent-id", true, nil)
+				_, _, err = fakeVMRepo.FindCurrentAgentId()
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedNetworkInterfaces = map[string]biproperty.Map{
+					"fake-network-name": biproperty.Map{
+						"type":             "dynamic",
+						"ip":               "10.10.1.3",
+						"cloud_properties": biproperty.Map{},
+						"default":          []bideplmanifest.NetworkDefault{"dns", "gateway"},
+					},
+				}
+			})
+
+			It("creates a VM with current IP specified", func() {
+				vm, err := manager.Create(stemcell, deploymentManifest)
+				Expect(err).ToNot(HaveOccurred())
+				expectedVM := NewVM(
+					"fake-vm-cid",
+					fakeVMRepo,
+					stemcellRepo,
+					fakeDiskDeployer,
+					fakeAgentClient,
+					fakeCloud,
+					clock.NewClock(),
+					fs,
+					logger,
+				)
+				Expect(vm).To(Equal(expectedVM))
+
+				Expect(fakeCloud.CreateVMInput).To(Equal(
+					fakebicloud.CreateVMInput{
+						AgentID:            "fake-existed-agent-id",
+						StemcellCID:        "fake-stemcell-cid",
+						CloudProperties:    expectedCloudProperties,
+						NetworksInterfaces: expectedNetworkInterfaces,
+						Env:                expectedEnv,
+					},
+				))
+
+				for _, networkInterface := range fakeCloud.CreateVMInput.NetworksInterfaces {
+					Expect(networkInterface["ip"]).To(Equal("10.10.1.3"))
+				}
+			})
```

Comment:
I'd would like to see some tests for failing paths... I know this is backfilling tests but since you are in it now, perhaps this is the correct time. Thx